spyderlib/plugins/__init__.py

```diff
@@ -55,6 +57,136 @@ def get_icon(self):
         return self.plugin.get_plugin_icon()
 
 
+class TabFilter(QObject):
+    """
+    Filter event attached to each QTabBar that holds 2 or more dockwidgets in
+    charge of handling tab rearangement.
+
+    This filter also holds the methods needed for the detection of a drag and
+    the movement of tabs.
+    """
+    sig_tab_moved = Signal(int, int)
+
+    def __init__(self, dock_tabbar, main):
+        QObject.__init__(self)
+        self.dock_tabbar = dock_tabbar
+        self.main = main
+        self.moving = False
+        self.from_index = None
+        self.to_index = None
+
+    # Helper methods
+    def _get_plugin(self, index):
+        """Get plugin reference based on tab index."""
+        for plugin in self.main.widgetlist:
+            if plugin.get_plugin_title() == self.dock_tabbar.tabText(index):
+                return plugin
+
+    def _get_plugins(self):
+        """
+        Get a list of all the plugins references in the QTabBar to which this
+        event filter was attached.
+        """
+        plugins = []
+        for index in range(self.dock_tabbar.count()):
+            plugin = self._get_plugin(index)
+            plugins.append(plugin)
+        return plugins
+
+    def _fix_cursor(self, from_index, to_index):
+        """Fix mouse cursor position to adjust for different tab sizes."""
+        direction = abs(to_index - from_index)/(to_index - from_index)
+        tab_width = self.dock_tabbar.tabRect(to_index).width()
+        tab_x_min = self.dock_tabbar.tabRect(to_index).x()
+        tab_x_max = tab_x_min + tab_width
+        previous_width = self.dock_tabbar.tabRect(to_index - direction).width()
+
+        delta = previous_width - tab_width
+        if delta > 0:
+            delta = delta * direction
+        else:
+            delta = 0
+        cursor = QCursor()
+        pos = self.dock_tabbar.mapFromGlobal(cursor.pos())
+        x, y = pos.x(), pos.y()
+        if x < tab_x_min or x > tab_x_max:
+            new_pos = self.dock_tabbar.mapToGlobal(QPoint(x + delta, y))
+            cursor.setPos(new_pos)
+
+    def eventFilter(self,  obj,  event):
+        """Filter mouse press events."""
+        event_type = event.type()
+        if event_type == QEvent.MouseButtonPress:
+            self.tab_pressed(event)
+            return False

```

Comment:
An event filter intercepts the fired events, and it has to return True, if it propagates the events... or False if it captures the event.... 

It is the normal behavior ....

http://doc.qt.io/qt-4.8/eventsandfilters.html#event-filters